ui/src/pages/feature/featureDetails.tsx

```diff
@@ -118,6 +123,85 @@ function InputDerivedFeatures(props: { project: string, feature: Feature }) {
   </>;
 }
 
+function FeatureLineageGraph() {
+  const [searchParams] = useSearchParams();
+  const { featureId } = useParams() as Params;
+
+  const [lineageData, setLineageData] = useState<FeatureLineage>({ guidEntityMap: null, relations: null });
+  const [elements, setElements] = useState<Elements>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    const fetchLineageData = async () => {
+      setLoading(true);
+      const data = await fetchFeatureLineages(featureId);
+      setLineageData(data);
+      setLoading(false);
+    };
+    
+    fetchLineageData();
+  }, [featureId]);
+
+  useEffect(() => {
+    const generateGraphData = async () => {
```

Comment:
>generateGraphData [](http://example.com/codeflow?start=10&length=17)

Can this function be reused? Currently this code is duplicated in both feature details and lineage page.